cdisc_rules_engine/services/cache/cache_populator_service.py

```diff
@@ -0,0 +1,251 @@
+        return variable_codelist_maps
+
+    async def _async_get_variable_codelist_map(
+        self, standard_type: str, standard_version: str
+    ) -> dict:
+        loop = asyncio.get_event_loop()
+        variables_map: dict = await loop.run_in_executor(
+            None,
+            self.library_service.get_variable_codelists_map,
+            standard_type,
+            standard_version,
+        )
+        return variables_map
+
+    async def _async_get_details_of_all_standards(
+        self, standards: List[dict]
+    ) -> List[dict]:
+        """
+        Gets details for each given standard.
+        """
+        coroutines = [
+            self._async_get_standard_details(
+                standard.get("href", "").split("/")[-2],
+                standard.get("href", "").split("/")[-1],
+            )
+            for standard in standards
+        ]
+        return await asyncio.gather(*coroutines)
+
+    async def _async_get_standard_details(
+        self, standard_type: str, standard_version: str
+    ) -> dict:
+        """
+        Gets details of a given standard.
+        """
+        loop = asyncio.get_event_loop()
+        standard_details: dict = await loop.run_in_executor(
+            None,
+            self.library_service.get_standard_details,
+            standard_type,
+            standard_version,
+        )
+        standard_details["cache_key"] = get_standard_details_cache_key(
+            standard_type, standard_version
+        )
+        return standard_details
+
+    async def _get_variables_metadata(self, standards: List[dict]) -> Iterable[dict]:
+        """
+        Returns a list of dicts of variables metadata for each standard.
+        """
+        coroutines = [
+            self._async_get_variables_metadata(
+                standard.get("href", "").split("/")[-2],
+                standard.get("href", "").split("/")[-1],
+            )
+            for standard in standards
+        ]
+        metadata = await asyncio.gather(*coroutines)
+        return filter(lambda item: item is not None, metadata)
+
+    async def _async_get_variables_metadata(
+        self, standard_type: str, standard_version: str
+    ) -> Optional[dict]:
+        """
+        Returns variables metadata for a given standard.
+        """
+        loop = asyncio.get_event_loop()
+        try:
+            variables_metadata: dict = await loop.run_in_executor(
+                None,
+                partial(
+                    self.library_service.get_variables_details,
+                    standard_type,
+                    standard_version,
+                ),
+            )
+        except LibraryResourceNotFoundException:
+            return None
+        return {
+            "cache_key": get_library_variables_metadata_cache_key(
+                standard_type, standard_version
+            ),
+            **variables_metadata,
+        }
+
+    def save_rules_locally(self, cache_path: str):
```

Comment:
Can the public methods be moved up so that they were declared before all encapsulated methods?